lib/matplotlib/offsetbox.py

```diff
@@ -350,8 +352,12 @@ def get_window_extent(self, renderer=None):
         # docstring inherited
         if renderer is None:
             renderer = self.figure._get_renderer()
-        w, h, xd, yd, offsets = self.get_extent_offsets(renderer)
-        px, py = self.get_offset(w, h, xd, yd, renderer)
+        w, h, xd, yd = self.get_extent(renderer)
+        # Some subclasses redefine get_offset to take no args.
+        if inspect.signature(self.get_offset).parameters:
+            px, py = self.get_offset(w, h, xd, yd, renderer)
+        else:
+            px, py = self.get_offset()
```

Comment:
Is this better than

```suggestion
        try:
            px, py = self.get_offset()
        except TypeError:
            px, py = self.get_offset(w, h, xd, yd, renderer)
```
?

Open question. I don't know the answer myself.

Reply:
In general I prefer the former because the latter can also hide a TypeError happening later in the code (not at argument binding time).  Here this is unlikely (because the two forms are close enough that if a TypeError occurs in the first case it will likely also occur in the second case), so I can switch to your form if you strongly prefer that.
Naturally using inspect.signature is not foolproof either (e.g. in the presence of `*args, **kwargs`)...

Reply:
The inspect way is somewhat slower (10 µs vs 400ns for a try-except). Not sure if that's relevant. Can this code be called in a hot loop, e.g. some interactivity where people are dragging things around?
This might be micro-optimization. Anyway, I have a slight preference for try-except, but won't insist.

Reply:
Done (favouring the with-args form, as that's the signature in that same class).